replication_timeout_test.go

```diff
@@ -285,3 +290,147 @@ func TestReplicationRequestIncompleteResponses(t *testing.T) {
 	laggingNode.e.AdvanceTime(laggingNode.e.StartTime.Add(simplex.DefaultReplicationRequestTimeout * 2))
 	laggingNode.storage.waitForBlockCommit(startSeq)
 }
+
+type collectNotarizationComm struct {
+	lock          *sync.Mutex
+	notarizations map[uint64]*simplex.Notarization
+	testNetworkCommunication
+
+	replicationResponses chan struct{}
+}
+
+func newCollectNotarizationComm(nodeID simplex.NodeID, net *inMemNetwork, notarizations map[uint64]*simplex.Notarization, lock *sync.Mutex) *collectNotarizationComm {
+	return &collectNotarizationComm{
+		notarizations:            notarizations,
+		testNetworkCommunication: newTestComm(nodeID, net, allowAllMessages),
+		replicationResponses:     make(chan struct{}, 3),
+		lock:                     lock,
+	}
+}
+
+func (c *collectNotarizationComm) Send(msg *simplex.Message, to simplex.NodeID) {
+	if msg.Notarization != nil {
+		c.lock.Lock()
+		c.notarizations[msg.Notarization.Vote.Round] = msg.Notarization
+		c.lock.Unlock()
+	}
+	c.testNetworkCommunication.Send(msg, to)
+}
+
+func (c *collectNotarizationComm) Broadcast(msg *simplex.Message) {
```

Comment:
you can just use the `recordingComm` we already have defined in `epoch_failover_test.go`. It fits what you're doing here perfectly.